src/rust/catmem/mod.rs

```diff
@@ -92,18 +92,36 @@ impl CatmemLibOS {
     pub fn open_pipe(&mut self, name: &str) -> Result<QDesc, Fail> {
         trace!("open_pipe() name={:?}", name);
 
-        let ring: SharedRingBuffer<u8> = SharedRingBuffer::<u8>::open(name, RING_BUFFER_CAPACITY)?;
+        let ring: SharedRingBuffer<u16> = SharedRingBuffer::<u16>::open(name, RING_BUFFER_CAPACITY)?;
 
         let qd: QDesc = self.qtable.alloc(QType::MemoryQueue.into());
         assert_eq!(self.rings.insert(qd, Rc::new(ring)).is_none(), true);
 
         Ok(qd)
     }
 
+    // Pushes EoF.
+    fn push_eof(&mut self, ring: Rc<SharedRingBuffer<u16>>) -> Result<(), Fail> {
+        let x: u16 = ((1 & 0xff) << 8) | (0 & 0xff) as u16;
```

Comment:
Likewise, I'd just do (1 << 8) as u16 here.  Not that it matters, the compiler will probably/hopefully process this whole expression into the equivalent constant at compile time.